packages/plugins/plugin-meeting/src/components/CallContainer.tsx

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import React, { useCallback, useEffect, type FC } from 'react';
+
+import { useCapability } from '@dxos/app-framework';
+import { invariant } from '@dxos/invariant';
+import { fullyQualifiedId } from '@dxos/react-client/echo';
+import { StackItem } from '@dxos/react-ui-stack';
+
+import { Call } from './Call';
+import { Lobby } from './Lobby';
+import { MeetingCapabilities } from '../capabilities';
+import { useCompanions } from '../hooks';
+import { type MeetingType } from '../types';
+
+export type CallContainerProps = {
+  meeting?: MeetingType;
+  roomId?: string;
+};
+
+export const CallContainer: FC<CallContainerProps> = ({ meeting, roomId: _roomId }) => {
+  const call = useCapability(MeetingCapabilities.CallManager);
+  const roomId = meeting ? fullyQualifiedId(meeting) : _roomId;
+  invariant(roomId);
```

Comment:
avoid invariants in react components main function (i.e., move inside hook callbacks)?